src/projects/TodoList/index.js

```diff
@@ -61,6 +83,23 @@ class TodoList extends React.Component {
     alteredTask.editingTask = value;
     this.setState({ todos: this.state.todos });
   }
+  toggleCheckBox(i) {
+    let alteredCheckBox = this.state.categories[i];
+    alteredCheckBox.checked = !alteredCheckBox.checked;
+    this.setState({ categories: this.state.categories });
+  }
+  addCategory() {
+    const largestCategoryId = this.state.categories[(this.state.categories).length-1].id
+    const newCategoryColour = "rgb(" + this.state.newCategoryRed + "," + this.state.newCategoryGreen + "," + this.state.newCategoryBlue + ")";
+    const newCategory = {};
+    newCategory.id = largestCategoryId+1
+    newCategory.category = this.state.newCategory
+    newCategory.colour = newCategoryColour
+    newCategory.checked = false
```

Comment:
Rather than creating a new empty object then assigning all its properties on a line by line basis, you can do this when you create the object:

```js
const newCategory = {
  id: latestCategoryId + 1,
  category: this.state.newCategory
  ...
};
```